src/rational/q/arithmetic/root.rs

```diff
@@ -0,0 +1,205 @@
+// Copyright © 2023 Sven Moog
+//
+// This file is part of qFALL-math.
+//
+// qFALL-math is free software: you can redistribute it and/or modify it under
+// the terms of the Mozilla Public License Version 2.0 as published by the
+// Mozilla Foundation. See <https://mozilla.org/en-US/MPL/2.0/>.
+
+//! This module provides implementations for calculating roots on [`Q`].
+//!
+//! # Max Error Calculation:
+//! Notation:
+//! - `a/b` is the "correct" result
+//! - `e_a` and `e_b` are the error of `a` and `b` resp.
+//! - `p` is the maximum error of a sqrt on [`Z`] values
+//!  => `|e_a| <= p` and `|e_b| <= p`
+//!
+//! ```Q::sqrt(x/y) = (a+e_a)/(b+e_b) = a/b + (e_a*b - a*e_b)/(b*(b+e_b))```
+//!
+//! The Error is the largest with `e_a = p` and `e_b = -p`:
+//! ```|(e_a*b - a*e_b)/(b*(b+e_b))| <= a/b * (b+1) * p/(b-p)```
```

Comment:
It is `(a/b) * (b+1) * (p/(b-p))` which is in my opinion clear enough, since that's how it should be evaluated based on the brackets. The additional spaces next to the `*` should make this even more clear.